File: NayifatAPI/Models/FinnoneModels.cs
using System;

namespace NayifatAPI.Models;

public class FinnoneCustomerRequest
{
    public string ApplicationNo { get; set; }
    public string NationalId { get; set; }
    public string FullNameEn { get; set; }
    public string FullNameAr { get; set; }
    public DateTime DateOfBirth { get; set; }
    public string Email { get; set; }
    public string Phone { get; set; }
    public decimal Salary { get; set; }
    public string ApplicationType { get; set; }  // LOAN or CARD
    public decimal RequestedAmount { get; set; }
    public int? Tenure { get; set; }            // For loans
    public string CardType { get; set; }        // For cards
}

public class FinnoneCustomerResponse
{
    public bool Success { get; set; }
    public string Message { get; set; }
    public string FinnoneReference { get; set; }
    public string Status { get; set; }
    public DateTime ProcessedAt { get; set; }
    public Dictionary<string, string> AdditionalData { get; set; }
}

public class FinnoneStatusResponse
{
    public string ApplicationNo { get; set; }
    public string FinnoneReference { get; set; }
    public string Status { get; set; }
    public string StatusDescription { get; set; }
    public DateTime LastUpdated { get; set; }
    public Dictionary<string, string> StatusDetails { get; set; }
}

// Common interface for both loan and card applications
public interface IApplication
{
    string ApplicationNo { get; set; }
    string NationalId { get; set; }
    string Type { get; }  // "LOAN" or "CARD"
    string Status { get; set; }
} 

File: lib/models/finnone_models.dart
class FinnoneCustomerResponse {
  final bool success;
  final String message;
  final String? finnoneReference;
  final String? status;
  final DateTime? processedAt;
  final Map<String, String>? additionalData;

  FinnoneCustomerResponse({
    required this.success,
    required this.message,
    this.finnoneReference,
    this.status,
    this.processedAt,
    this.additionalData,
  });

  factory FinnoneCustomerResponse.fromJson(Map<String, dynamic> json) {
    return FinnoneCustomerResponse(
      success: json['success'] ?? false,
      message: json['message'] ?? '',
      finnoneReference: json['finnoneReference'],
      status: json['status'],
      processedAt: json['processedAt'] != null 
          ? DateTime.parse(json['processedAt']) 
          : null,
      additionalData: json['additionalData'] != null 
          ? Map<String, String>.from(json['additionalData'])
          : null,
    );
  }
}

class FinnoneStatusResponse {
  final String applicationNo;
  final String? finnoneReference;
  final String status;
  final String statusDescription;
  final DateTime lastUpdated;
  final Map<String, String>? statusDetails;

  FinnoneStatusResponse({
    required this.applicationNo,
    this.finnoneReference,
    required this.status,
    required this.statusDescription,
    required this.lastUpdated,
    this.statusDetails,
  });

  factory FinnoneStatusResponse.fromJson(Map<String, dynamic> json) {
    return FinnoneStatusResponse(
      applicationNo: json['applicationNo'] ?? '',
      finnoneReference: json['finnoneReference'],
      status: json['status'] ?? '',
      statusDescription: json['statusDescription'] ?? '',
      lastUpdated: DateTime.parse(json['lastUpdated']),
      statusDetails: json['statusDetails'] != null 
          ? Map<String, String>.from(json['statusDetails'])
          : null,
    );
  }
} 